### PR TITLE
Trash manager

### PR DESCRIPTION
**Pull request by alroniks**
See original PR here https://github.com/modxcms/revolution/pull/13764 for details.

This is rebased version with additional improvements. Original will be closed after finish if work on improvements.

## Template build required
The PR needs to run `_build/templates/default/grunt build`. Otherwise the hover state of the buttons is not visible and the changes in modx.tree.resource.js are not in the compressed scripts.

### CODE REVIEW

File: core/model/modx/processors/resource/trash/restore.class.php [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+                $this->resource->removeLock();
+
+                $this->failures[] = $id;
+
+                return $this->failure($this->modx->lexicon('resource_err_undelete'));
+            } else {
+                $this->success[] = $id;
+            }
+
+            // TODO this still has to be discussed: what happens to the children?
```
Comment (JoshuaLuckers) [inline core/model/modx/processors/resource/trash/restore.class.php]: Issue #11973 might be sort of relevant here

**Comment by Jako** [comment]
Checked, refactored, working fine.

**Review by Jako — DISMISSED** [review]

**Comment by alroniks** [comment]
I also want to do some refactoring in grid component, but right now I have a lack of time. So I removed state `ready-for-merging`

**Review by Mark-H — APPROVED** [review]
While it looks like there's still work being done, I'll go ahead and approve the current state. Looks really promising.

**Review by Jako — APPROVED** [review]

**Comment by opengeek** [comment]
Is this ready for merge or not? It's approved for it but says it needs review and @Mark-H mentioned it looks like work is still ongoing? @Alroniks ?

**Comment by alroniks** [comment]
I wanted to improve some in code, fix some smelling places, but it works well and can be merged, all further improvements I can add later. So feel free to merge. I don't want to block a release.

**Pull request merged**